reconciling fork

alexwennerberg opened the pull request:
cleaned up some of my changes so each one is in a separate commit, should be relatively clean and atomic. feel free to cherry pick them or merge this branch, lmk what works best for you or if I should split up this work into separate PRs, and let me know if you have any feedback about these changes!

cblgh commented on server/server.go:
```diff
@@ -286,6 +293,18 @@ func (h RequestHandler) renderGenericMessage(res http.ResponseWriter, req *http.
 	return
 }
 
+func (h RequestHandler) renderGeneric401(res http.ResponseWriter, req *http.Request) {
+	res.WriteHeader(401)
+	title := h.translator.Translate("ErrGeneric401")
+	data := GenericMessageData{
+		Title:   title,
+		Message: h.translator.Translate("ErrGeneric401Message"),
+	}
+	h.renderGenericMessage(res, req, data)
+	return
+
+}
+
```
this is only used in one place and it's very specific, inline it instead? if the pattern repeats of needing to handle multiple error codes generically then this pattern could be reused by also accepting a status code

alexwennerberg replied:
ie, make it `fmt.Sprintf(ErrGeneric%d)`?

alexwennerberg replied:
inlined it for now!

cblgh commented on server/server.go:
```diff
@@ -791,6 +800,35 @@ func (h *RequestHandler) DeletePostRoute(res http.ResponseWriter, req *http.Requ
+		h.db.EditPost(content, postid)
+		post.Content = content
+	}
+	// update rss feed
+	h.rssFeed = GenerateRSS(h.db, h.config)
```
probably not needed / potentially confusing based on how the feed is generated rn? it basically takes the last post of each thread as the "latest post", linking to it and saying something like "[date] [person] posted"

alexwennerberg replied:
ah ya I didnt look to close on this. I'll change it!

cblgh commented:
hey thanks! :) i really appreciate this reconciling PR a bunch and your effort in having squashed the separate additions into individual, atomic commits

in the future, could you yank out the commit running the `go fmt`? per the [contributing guidelines](https://github.com/cblgh/cerca/blob/main/CONTRIBUTING.md):

> Don't go fmt the entire codebase in the same PR as you're adding a feature; do that separately if it's needed

it reduces the burden in reviewing many irrelevant lines, which can induce mistakes and oversights. i understand that a lot of the code that needed go fmt were mine but the practice remains the same

---

i gotta mull over replies :) i have concerns about the quote pyramids from ye olden times, as well as how replies compose with the edit functionality also introduced in this pr. gotta say i like that you landed on a no-js solution for it tho :>

also still not a fan of relative timestamps in the default templates and will probably replace them for the current timestamps after merging (but i do like the addition of the relativeDate template function!)

alexwennerberg commented:
> hey thanks! :) i really appreciate this reconciling PR a bunch and your effort in having squashed the separate additions into individual, atomic commits
> 
> in the future, could you yank out the commit running the `go fmt`? per the [contributing guidelines](https://github.com/cblgh/cerca/blob/main/CONTRIBUTING.md):
> 
> > Don't go fmt the entire codebase in the same PR as you're adding a feature; do that separately if it's needed
>
> it reduces the burden in reviewing many irrelevant lines, which can induce mistakes and oversights. i understand that a lot of the code that needed go fmt were mine but the practice remains the same
ah, my bad, I missed that. I have my editor configured to run `go fmt` on save, I'll disable that and redo this pr
> 
> i gotta mull over replies :) i have concerns about the quote pyramids from ye olden times, as well as how replies compose with the edit functionality also introduced in this pr. gotta say i like that you landed on a no-js solution for it tho :>
Thanks! page loads are so fast that there's basically no difference from a UX perspective too. It's been fun trying to think about how to add typical forum features without JS

> also still not a fan of relative timestamps in the default templates and will probably replace them for the current timestamps after merging (but i do like the addition of the relativeDate template function!)
You mean `formatDateTimeRelative`? I don't think I introduced any usage of this function in this PR?

I'll make the changes you requested later this morning!

alexwennerberg commented:
updated with discussed changes :)

cblgh commented:
woop! i'll have a run through of the PR on my local env tomorrow and then merge :~

> You mean `formatDateTimeRelative`? I don't think I introduced any usage of this function in this PR?

my bad, i misread the function names :~

cblgh commented:
most of this was merged in #54, closin
